client/starwhale/api/_impl/dataset/model.py

```diff
@@ -159,6 +159,9 @@ def __init__(
         self._writer_lock = threading.Lock()
         self._row_writer: t.Optional[RowWriter] = None
         self.__keys_cache: t.Set[t.Union[int, str]] = set()
+        if self._create_by_append:
+            loader = get_data_loader(_origin_uri)
+            self.__keys_cache = set([row.id for row in loader.tabular_dataset.scan()])
```

Comment:
it will be too slow to load all data?

Reply:
Another point, it will make the init process slower. A lazy load is the better choice.